src/Text/Pandoc/Readers/JATS.hs

```diff
@@ -273,26 +273,27 @@ parseBlock (Elem e) =
                                            Just c -> filterChildren isColspec c
                                            _      -> filterChildren isColspec e'
                       let isRow x = named "row" x || named "tr" x
-                      headrows <- case filterChild (named "thead") e' of
-                                       Just h  -> case filterChild isRow h of
-                                                       Just x  -> parseRow x
-                                                       Nothing -> return []
-                                       Nothing -> return []
-                      bodyrows <- case filterChild (named "tbody") e' of
-                                       Just b  -> mapM parseRow
-                                                  $ filterChildren isRow b
-                                       Nothing -> mapM parseRow
-                                                  $ filterChildren isRow e'
+
+                      -- list of header cell elements
+                      let headRowElements = case filterChild (named "thead") e' of
+                                      Just h -> maybe [] parseElement (filterChild isRow h)
+                                      Nothing -> []
+                      -- list of list of body cell elements 
+                      let bodyRowElements = case filterChild (named "tbody") e' of
+                                      Just b -> map parseElement $ filterChildren isRow b
+                                      Nothing -> map parseElement $ filterChildren isRow e'
                       let toAlignment c = case findAttr (unqual "align") c of
                                                 Just "left"   -> AlignLeft
                                                 Just "right"  -> AlignRight
                                                 Just "center" -> AlignCenter
                                                 _             -> AlignDefault
+                      let toColSpan element =  fromMaybe 1 $ safeRead $ fromMaybe "1" $ findAttr (unqual "colspan") element
+                      let toRowSpan element =  fromMaybe 1 $ safeRead $ fromMaybe "1" $ findAttr (unqual "rowspan") element
```

Comment:
We try to keep lines to 80 columns or less.
In this case, you could probably avoid having two `fromMaybe`s:
```
fromMaybe 1 $ findAttr (unqual "colspan") element >>= safeRead
```
(untested). similarly on next line.